sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -172,55 +173,64 @@ private static ErrorDetails CreateExceptionDetails(Response response, RequestFai
 
-            if (!string.IsNullOrEmpty(response.ReasonPhrase))
+            if (response.IsError)
             {
-                messageBuilder.Append(" (")
-                    .Append(response.ReasonPhrase)
-                    .AppendLine(")");
-            }
-            else
-            {
-                messageBuilder.AppendLine();
-            }
+                if (!string.IsNullOrEmpty(response.ReasonPhrase))
+                {
+                    messageBuilder.Append(" (")
+                        .Append(response.ReasonPhrase)
+                        .AppendLine(")");
+                }
+                else
+                {
+                    messageBuilder.AppendLine();
+                }
 
-            if (!string.IsNullOrWhiteSpace(error?.Code))
-            {
-                messageBuilder.Append("ErrorCode: ")
-                    .Append(error?.Code)
-                    .AppendLine();
-            }
+                if (!string.IsNullOrWhiteSpace(error?.Code))
+                {
+                    messageBuilder.Append("ErrorCode: ")
+                        .Append(error?.Code)
+                        .AppendLine();
+                }
 
-            if (additionalInfo is { Count: > 0 })
-            {
-                messageBuilder
-                    .AppendLine()
-                    .AppendLine("Additional Information:");
-                foreach (KeyValuePair<string, string> info in additionalInfo)
+                if (additionalInfo is { Count: > 0 })
                 {
                     messageBuilder
-                        .Append(info.Key)
-                        .Append(": ")
-                        .AppendLine(info.Value);
+                        .AppendLine()
+                        .AppendLine("Additional Information:");
+                    foreach (KeyValuePair<string, string> info in additionalInfo)
+                    {
+                        messageBuilder
+                            .Append(info.Key)
+                            .Append(": ")
+                            .AppendLine(info.Value);
+                    }
+                }
+
+                if (response.ContentStream is MemoryStream && ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out Encoding _))
+                {
+                    messageBuilder
+                        .AppendLine()
+                        .AppendLine("Content:")
+                        .AppendLine(response.Content.ToString());
                 }
-            }
 
-            if (response.ContentStream is MemoryStream && ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out Encoding _))
-            {
                 messageBuilder
                     .AppendLine()
-                    .AppendLine("Content:")
-                    .AppendLine(response.Content.ToString());
-            }
+                    .AppendLine("Headers:");
```

Comment:
Curious: is it a security risk to include sanitized headers in the exception message?  I don't feel strongly one way or another, but my initial understanding of the proposal was that only the response body would be omitted from the exception message, so just looking to get my head around the delta from that mental model.

Reply:
Unless you feel strongly, I prefer to redact everything but the status. The raw response is always available to inspect if needed.

Reply:
I don't feel strongly, and happy to defer to you -- and also more conservative/safer approaches in general -- on security topics :)